spec/solver_spec.rb

```diff
@@ -0,0 +1,25 @@
+require_relative './spec_helper'
+
+describe Solver do
+  context 'When testing the Solver class' do
+    num = 5
+    word = 'maano'
+    solver = Solver.new
+
+    it 'solver should indeed have method factorial' do
+      expect(solver).to respond_to(:factorial)
+    end
+
+    it 'factorial of 5 should be 120' do
+      expect(solver.factorial(num)).to eq 120
+    end
+
+    it 'reverse of maano should be onaam' do
+      expect(solver.reverse(word)).to match(/^onaam+$/)
+    end
```

Comment:
- [ ] kindly write a factorial test that handles a special case when the input is 0

Reply:
I will implement these changes as well 🚀.